pxp/versions/1.1/request_response.md

```diff
@@ -0,0 +1,268 @@
+Request/Response transactions
+===
+
+In this section, we describe the request/response message exchange between
+client endpoints.
+
+### Blocking Action - Message Flow
+
+The following figure shows an example of a *blocking* request/response
+transaction, for a `puppet status` action.
+
+```
+    controller C                agent A
+       |                           |
+       |  `puppet status` request  |
+     1 |-------------------------->|
+       |                           | puppet status 2
+       |                           |   |
+       |                           |   |
+       |                           |   |
+       |          response         |   V 3
+     4 |<--------------------------|
+       |                           |
+```
+
+The controller **C** and the agent **A** are registered in the same PCP
+framework. **C** sends a `puppet status` *blocking request* to **A** (1).
+As soon as the request is received and validated, **A** starts the execution of
+`puppet status` by relying on the logic of the puppet module (2).
+
+Once the action terminates (3), **A** sends back to **B** a *blocking response*
+containing the `puppet status` outcome (4).
+
+Note that the transaction does not employ acknowledgment messages. And there
+are no constraints regarding the controller operation; **C** can decide whether
+or to not to implement a response timeout.
+
+Also, note that **A** must reply with a:
+ - *PCP protocol error message* in case the request message contains a bad data chunk (see [PCP Error Handling][1])
```

Comment:
It would be if the target was the controller. I haven't changed this, and not aware of any issues with it.